src/main/webapp/app/exercises/programming/manage/services/programming-exercise.service.ts

```diff
@@ -475,6 +476,38 @@ export class ProgrammingExerciseService {
         return this.http.get<ProgrammingExerciseFullGitDiffReport>(`${this.resourceUrl}/${exerciseId}/full-diff-report`);
     }
 
+    /**
+     * Gets the testwise coverage report of a programming exercise for the latest solution submission with all descending reports
+     * @param exerciseId The id of a programming exercise
+     */
+    getLatestFullTestwiseCoverageReport(exerciseId: number): Observable<CoverageReport> {
+        return this.http.get<CoverageReport>(`${this.resourceUrl}/${exerciseId}/full-testwise-coverage-report`);
+    }
+
+    /**
+     * Gets the testwise coverage report of a programming exercise for the latest solution submission without the actual reports
+     * @param exerciseId The id of a programming exercise
+     */
+    getLatestTestwiseCoverageReport(exerciseId: number): Observable<CoverageReport> {
+        return this.http.get<CoverageReport>(`${this.resourceUrl}/${exerciseId}/full-testwise-coverage-report`);
+    }
+
+    /**
+     * Gets all files from the last solution participation repository
+     */
+    getSolutionRepositoryTestFilesWithContent(exerciseId: number): Observable<Map<string, string> | undefined> {
+        return this.http.get(`${this.resourceUrl}/${exerciseId}/solution-files-content`).pipe(
+            map((res: HttpResponse<any>) => {
+                // this mapping is required because otherwise the HttpResponse object would be parsed
+                // to an arbitrary object (and not a map)
+                if (res === undefined) {
+                    return undefined;
+                }
+                return new Map(Object.entries(res));
```

Comment:
I think this could be simplified
```suggestion
                return res && new Map(Object.entries(res));
```